_docs-sources/pipelines/tutorial/index.md

```diff
@@ -1,19 +1,358 @@
+```shell
+docker push "$AWS_ACCOUNT_ID.dkr.ecr.$AWS_REGION.amazonaws.com/ecs-deploy-runner:$TERRAFORM_AWS_CI_VERSION"
+```
+
+```shell
+docker push "$AWS_ACCOUNT_ID.dkr.ecr.$AWS_REGION.amazonaws.com/kaniko:$TERRAFORM_AWS_CI_VERSION"
+```
+
+## Step 4: Deploy the Pipelines Cluster
+
+Now that the ECR repositories are deployed and have the required docker images, we are ready
+to deploy the rest of Gruntwork Pipelines. The terraform that defines the setup is defined in
+`terraform-aws-service-catalog/examples/for-learning-and-testing/gruntwork-pipelines/pipelines-cluster`
+
+```shell
+cd terraform-aws-service-catalog/examples/for-learning-and-testing/gruntwork-pipelines/pipelines-cluster
+```
+
+### Step 4a: Export a GitHub Personal Access Token (PAT)
+For the purposes of this example, you may use the same PAT as in step 3a. In a production deployment, best practice
+would be to create a separate GitHub machine user account. This modules uses a slightly different naming convention for
+its environment variable so you'll need to re-export the token:
+
+```shell
+export TF_VAR_github_token=<your-github-personal-access-token>
+```
+
+### Step 4b: Configure and Deploy the ecs deploy runner
+Authenticate to your AWS account and run init, then apply.
+:::note
+If you  are using aws-vault to authenticate on the command line, you must supply the `--no-session` flag as explained in [this KB entry](https://github.com/gruntwork-io/knowledge-base/discussions/647)
+:::
+
+```shell
+terraform init
+```
+
+```shell
+terraform plan
+```
+Check your plan output before applying
+```shell
+terraform apply
+```
+
+## Step 5: Install the `infrastructure-deployer` command line tool
+
+Gruntwork Pipelines requires all requests to transit through its fronting Lambda function, which ensures only valid arguments and commands are passed along to ECS.
+To invoke Gruntwork Pipelines's fronting Lambda function, you should use the `infrastructure-deployer` command line interface (CLI) tool. For testing and setup purposes, we'll install and use the `infrastructure-deployer` CLI locally; when you're ready to configure CI / CD, you'll install and use it in your CI / CD config.
+
+If you do not already have the `gruntwork-install` binary installed, you can get it [here.](https://github.com/gruntwork-io/gruntwork-installer)
+
+```bash
+
+gruntwork-install --binary-name "infrastructure-deployer" --repo "https://github.com/gruntwork-io/terraform-aws-ci" --tag "$TERRAFORM_AWS_CI_VERSION"
+```
+:::note
+If you'd rather not use the Gruntwork installer, you can alternatively download the binary manually from [the releases page.](https://github.com/gruntwork-io/terraform-aws-ci/releases)
+:::
+
+## Step 6: Invoke your Lambda Function
+
+### Step 6a: Get your Lambda ARN from the output
+There is a Terraform output, `gruntwork_pipelines_lambda_arn`, configured that will tell you the ARN of the fronting Lambda that
```

Comment:
```suggestion
Next, we need to retrieve the Amazon Resource Name (ARN) for the Lambda function that guards your Gruntwork Pipelines installation.
```

I think it's a safe assumption that they missed it during apply time, so maybe we just say "retrieve it using the following command"?